iceoryx_utils/include/iceoryx_utils/internal/lockfree_queue/unique.hpp

```diff
@@ -0,0 +1,92 @@
+#pragma once
+
+#include <utility>
+
+namespace iox
+{
+template <typename T>
+class unique
+{
+  public:
+    struct invalid_t
+    {
+        explicit constexpr invalid_t()
+        {
+        }
+    };
+
+    static constexpr invalid_t invalid{};
+
+    template <typename... Args>
+    unique(Args&&... args)
+        : m_value(std::forward<Args>(args)...)
+    {
+    }
+
+    unique()
```

Comment:
Isn't this ctor ambiguous to the variadic template argument constructor? Are they both tested in unit tests?

Reply:
unique is currently not tested separately but only in its use UniqueIndex. The ctor is redundant. I considered implementing it completely in UniqueIndex itself, but I think it makes more sense to derive from some common base class and adapt the functionality in UniqueIndex a little (mainly to limit construction to the IndexQueue itself, otherwise I would have used unique directly).